src/pages/ReportSettingsPage.js

```diff
@@ -116,19 +117,17 @@ class ReportSettingsPage extends Component {
             return errors;
         }
 
-        // The following validations are ordered by precedence.
-        // First priority: We error if the user doesn't enter a room name or left blank
         if (!values.newRoomName || values.newRoomName === CONST.POLICY.ROOM_PREFIX) {
-            errors.newRoomName = this.props.translate('newRoomPage.pleaseEnterRoomName');
-        } else if (ValidationUtils.isReservedRoomName(values.newRoomName)) {
-            // Second priority: Certain names are reserved for default rooms and should not be used for policy rooms.
-            errors.newRoomName = this.props.translate('newRoomPage.roomNameReservedError');
-        } else if (ValidationUtils.isExistingRoomName(values.newRoomName, this.props.reports, this.props.report.policyID)) {
-            // Third priority: Show error if the room name already exists
-            errors.newRoomName = this.props.translate('newRoomPage.roomAlreadyExistsError');
-        } else if (!ValidationUtils.isValidRoomName(values.newRoomName)) {
-            // Fourth priority: We error if the room name has invalid characters
-            errors.newRoomName = this.props.translate('newRoomPage.roomNameInvalidError');
+            // We error if the user doesn't enter a room name or left blank
+            ErrorUtils.addErrorMessage(errors, 'newRoomName', this.props.translate('newRoomPage.pleaseEnterRoomName'));
+        } else if (values.newRoomName !== CONST.POLICY.ROOM_PREFIX && !ValidationUtils.isValidRoomName(values.newRoomName)) {
+            // We error if the room name has invalid characters
+            ErrorUtils.addErrorMessage(errors, 'newRoomName', this.props.translate('newRoomPage.roomNameInvalidError'));
+        }
+
+        if (ValidationUtils.isReservedRoomName(values.newRoomName) || ValidationUtils.isExistingRoomName(values.newRoomName, this.props.reports, values.policyID)) {
```

Comment:
As I mentioned on the issue, `values.policyID` can be undefined so lets get it from the report.

Also, we should keep the validation / messages for reserved room names and existing room names separate. I may have mislead you slightly on the issue. We had very similar error messages "A room on this workspace already uses this name" and "A room with this name already exists". The former is for a reserved room name like #admins or #announce, so we should actually update the error message to be more clear that these names are not allowed. We could follow a similar pattern as when using "Concierge" or "Expensify" in your first name.